src/lib/transaction_snark/transaction_snark.ml

```diff
@@ -2235,6 +2235,10 @@ module Make_str (A : Wire_types.Concrete) = struct
       let fee_payer =
         Account_id.Checked.create payload.common.fee_payer_pk fee_token
       in
+      let%bind () =
+        [%with_label_ "Fee-payer and soure must be equal"] (fun () ->
+            Account_id.Checked.equal fee_payer source >>= Boolean.Assert.is_true )
+      in
```

Comment:
Why do we need to check that here if the check is already present in the transaction logic?

Reply:
This is the snark code for the transaction union snark (used for signed commands). Unlike the zkapps command logic, it is implemented separately from the out-of-snark implementation, so we have to update it in both places.

Reply:
I see. Thanks for explanation.